cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1050,10 +1050,7 @@ private Future<KafkaRebalanceStatus> reconcileRebalance(Reconciliation reconcili
             patchedKafkaRebalance
                     .editMetadata()
                          .addToAnnotations(Map.of(ANNO_STRIMZI_IO_REBALANCE, KafkaRebalanceAnnotation.refresh.toString()))
-                    .endMetadata()
-                    .editStatus()
-                         .withObservedGeneration(kafkaRebalance.getMetadata().getGeneration())
-                    .endStatus();
+                    .endMetadata();
```

Comment:
in the PR description you say "It also removes the unused status section when the refresh annotation is added", what do you mean by "unused" here?

Reply:
Good point. You are right. I would just say, let's keep this out from this PR because unrelated and open a really trivial one with this patch. Thanks!

Reply:
Hope you are happy now because I'm quite close to just stop cleaning the mess of `KafkaRebalanceAssemblyOperator` and get back to restart the system tests twice per day as it seems much easier!

Reply:
> Hope you are happy now because I'm quite close to just stop cleaning the mess of KafkaRebalanceAssemblyOperator

I don't think it's fair. There are people working on this and I am not sure it could be called a mess. But feel free to rewrite it from scratch, we'll be happy to review your better code. Also I don't think I asked for big changes, I asked questions to understand those changes. Thanks!